packages/go_router/test/go_route_test.dart

```diff
@@ -249,6 +249,42 @@ void main() {
       expect(tester.takeException(), isAssertionError);
     });
 
+    testWidgets('throw if redirect to itself.', (WidgetTester tester) async {
+      final GoRouter router = await createRouter(
+        <RouteBase>[
+          GoRoute(
+            path: '/',
+            builder: (_, __) => const Text('home'),
+            routes: <RouteBase>[
+              GoRoute(
+                path: 'route',
+                name: 'route', // Named route
+                redirect: (_, __) => '/route',
+                routes: <RouteBase>[
+                  GoRoute(
+                    path: '1',
+                    builder: (_, __) => const Text('/route/1/2'),
+                  ),
+                ],
+              ),
+            ],
+          ),
+        ],
+        tester,
+      );
+      expect(find.text('home'), findsOneWidget);
+
+      // Test namedLocation with fragment
+      final String locationWithFragment =
+          router.namedLocation('route', fragment: '2');
+      expect(locationWithFragment, '/route#2');
+      // Navigate using goNamed with fragment
+      router.goNamed('route', fragment: '2');
+      await tester.pumpAndSettle();
+      // Should redirect to /route/1 without error.
+      expect(tester.takeException(), isAssertionError);
```

Comment:
why would this throw? I thought we allow redirect to the same route, which go_router treats it as no redirect.

Reply:
Sorry about it . You are right we should not expect isAssertionError . I haved updated the testCase to properly test the fragment handling in go-route